### ISSUE TITLE
Auth Providers: Add `search_using_service_account` field

### ISSUE THREAD

**Issue by rak-phillip** [rancher/dashboard#12871]
Add the ability to configure `searchUsingServiceAccount` while creating and editing openldap and freeipa auth providers. 

## Acceptance criteria

- [ ] `searchUsingServiceAccount` is able to be configured for openldap and freeipa auth providers
- [ ] `searchUsingServiceAccount` is a boolean with a default value of `false`

## Notes

- The new `searchUsingServiceAccount` setting is located under `/v3/authconfigs`
- `searchUsingServiceAccount` applies only to freeipa an openldap providers
- `shell/edit/auth/ldap/config.vue` is most likely the component that will receive this change
- The save method appears to be defined in `shell/mixins/auth-config.js`

## Details

This feature request is related to rancher/rancher#43064. 

Rancher currently scopes searches to the permissions of the requesting user to prevent permission escalation. This approach doesn't support typical setups where users have no permissions in the authentication provider, and a service account is used to control read permissions. Adding `search_using_service_account` allows users who use the more typical setups to use Rancher in the method that they desire.

**Comment by cnotv** [comment]
While setting up the auth provider I got this message and confused me a lot.

<img width="940" alt="Image" src="https://github.com/user-attachments/assets/2e51176d-fa10-4479-9cbc-aaec59db1347" />

**Comment by cnotv** [comment]
Also this icon not sure what means here

<img width="165" alt="Image" src="https://github.com/user-attachments/assets/0a6c5d99-d77c-4172-87db-2cfa1d14341f" />

**Comment by cnotv** [comment]
To add context, the search which needs to be enabled with the checkbox/toggler is inside the `Cluster Membership` configuration.
Here an example like in GitHub when searching an user to add:

<img width="1411" alt="Image" src="https://github.com/user-attachments/assets/38b7e948-2099-41ca-a1c0-b5d9ca36a7b7" />

**Comment by yonasberhe23** [comment]
Manual validation is required. While the related unit test verifies the existence of the configuration, we should perform regression testing around FreeIPA and OpenLDAP auth providers (e.g. ensure the configuration is correctly passed in the request payload).

**Comment by cnotv** [comment]
Yeah, I am trying to debug https://github.com/rancher/dashboard/issues/12477 and we do not have a single test in the whole authentication process, so it's a good idea to check if existing logic is still working, beside I did not touch the code, so it should.

**Comment by izaac** [comment]
Rancher version: `v2.11-9b286b6f24221c1834ed777a27fd52faef31daca-head`
UI version: `master 8f9913c`

FreeIPA and LDAP searches on the dropdown of the Auth Provider access of users and groups to Rancher Manager is working as expected. Adding, removing users and groups.
The dropdown search of Cluster Members worked as expected too. With the same CRUD checks.

**Comment by rak-phillip** [comment]
@gaktive We have confirmation from Collie that this can move forward. Do we want to target the 2.12 milestone or 2.13 at this point?

**Comment by gaktive** [comment]
QA has blessed this for 2.12.0.

**Comment by izaac** [comment]
This is goes straight to manual testing.

**Comment by izaac** [comment]
Rancher version: `v2.12-ada6550dbcad4b05c7f828d7f874b7d64b2771c1-head`
Rancher UI version: `master 88a9d18`
Providers: FreeIPA / OpenLDAP

Leaving the checkbox unmarked sends the json key with value false.

Clicking on the checkbox and completing enabling the Auth Provider sets the `searchUsingServiceAccount` value to `true` as expected:

Json Payload Excerpt:

json
```
        "baseType": "authConfig",
        "connectionTimeout": 5000,
        "created": "2025-07-17T15:41:56Z",
        "createdTS": 1752766916000,
        "creatorId": null,
        "enabled": true,
        "groupDNAttribute": "entrydn",
        "groupMemberMappingAttribute": "member",
        "groupMemberUserAttribute": "entrydn",
        "groupNameAttribute": "cn",
        "groupObjectClass": "groupOfNames",
        "groupSearchAttribute": "cn",
        "id": "freeipa",
        "labels": {
            "cattle.io/creator": "norman"
        },
        "links": {
            "self": "https://<>/v3/freeIpaConfigs/freeipa",
            "update": "https://<>/v3/freeIpaConfigs/freeipa"
        },
        "logoutAllSupported": false,
        "name": "freeipa",
        "port": 389,
        "searchUsingServiceAccount": true, //  <-------------- TRUE HERE
        "starttls": false,
        "status": {
            "conditions": null,
            "type": "/v3/schemas/authConfigStatus"
        },
```
